src/System.Slices/src/System/SpanEnumerators.cs

```diff
@@ -88,13 +96,7 @@ public void Dispose()
 
             public bool MoveNext()
             {
-                int nextItemIndex = _position + 1;
-                if (nextItemIndex < _span.Length)
-                {
-                    _position = nextItemIndex;
-                    return true;
-                }
-                return false;
+                return ++_position < _span.Length;
```

Comment:
@VSadov 
Hmm.. Never thought about the difference.
I will fix that in another PR.